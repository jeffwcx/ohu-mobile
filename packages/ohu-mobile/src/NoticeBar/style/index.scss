@use '../../_styles/variables' as *;
@use "pkg:@ohu-mobile/styles" as *;
@use '../../_styles/mixins' as *;

$noticeBarPrefixClass: #{$prefix}notice-bar;

// @keyframes notice-bar-scroll-animation {
//   0% {
//     transform: translateX(0);
//   }
//   100% {
//     transform: translateX(-100%);
//   }
// }

.#{$noticeBarPrefixClass} {
  box-sizing: border-box;
  position: relative;
  padding: $noticeBarPadding 30px;
  font-size: $noticeBarFontSize;
  &.is-default {
    background: $noticeBarDefaultBackground;
    color: $noticeBarDefaultColor;
  }
  &.is-warning {
    background: $noticeBarWarningBackground;
    color: $noticeBarWarningColor;
  }
  &.has-icon {
    padding-left: $noticeBarPadding + $noticeBarIconSize + 22px;
  }
  &.has-action {
    padding-right: $noticeBarPadding + $noticeBarIconSize + 22px;
  }
  &__icon {
    font-size: 1.4em;
    @include center(y);
    left: $noticeBarPadding;
    width: $noticeBarIconSize;
    height: $noticeBarIconSize;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  &__text {
    line-height: 1.43em;
    &.is-wrap div {
      @include text-wrap;
    }
    &.is-inline {
      overflow: hidden;
      div {
        white-space: nowrap;
      }
    }
    div {
      display: inline-block;
    }
  }
  &__action {
    font-size: 1.4em;
    @include center(y);
    right: $noticeBarPadding;
    width: $noticeBarIconSize;
    height: $noticeBarIconSize;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
