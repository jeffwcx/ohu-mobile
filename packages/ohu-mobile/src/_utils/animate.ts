import { EasingFunction } from './easing';
import raf from 'raf';

export interface AnimateOptions {
  timingFunction: EasingFunction;
  duration: number;
  done?: () => void;
  props: {
    height?: number | [number, number];
    width?: number | [number, number];
    scaleX?: number | [number, number];
    scaleY?: number | [number, number];
    scaleZ?: number | [number, number];
    // to be expanded
  };
}

const getPropChangeFunc = (
  initProp: number,
  endPropValue: number,
  execFunc: (value: number) => void,
) => {
  return (progress: number) => {
    const d = endPropValue - initProp;
    let p = Math.abs(d) * progress;
    if (d < 0) {
      p = initProp - p;
    }
    return execFunc(p);
  };
};

export default function animate(
  el: HTMLElement,
  { timingFunction, duration, done, props }: AnimateOptions,
) {
  let distanceArr: ((progress: number) => void)[] = [];
  if (props.width !== undefined) {
    let initValue;
    let endValue;
    if (typeof props.width === 'number') {
      const { width } = el.getBoundingClientRect();
      initValue = width;
      endValue = props.width;
    } else {
      const [i, e] = props.width;
      initValue = i;
      endValue = e;
    }
    distanceArr.push(
      getPropChangeFunc(initValue, endValue, (value) => {
        el.style.width = `${value}px`;
      }),
    );
  }
  if (props.height !== undefined) {
    let initValue;
    let endValue;
    if (typeof props.height === 'number') {
      const { height } = el.getBoundingClientRect();
      initValue = height;
      endValue = props.height;
    } else {
      const [i, e] = props.height;
      initValue = i;
      endValue = e;
    }
    distanceArr.push(
      getPropChangeFunc(initValue, endValue, (value) => {
        el.style.height = `${value}px`;
      }),
    );
  }
  let currentTime = Date.now();
  let tick = () => {
    const interval = Date.now() - currentTime;
    const execFunc = () => {
      const distanceProgress = timingFunction(interval / duration);
      distanceArr.forEach((func) => {
        func(distanceProgress);
      });
    };
    if (interval >= duration) {
      execFunc();
      done && done();
      return;
    }
    execFunc();
    raf(tick);
  };
  raf(tick);
}
