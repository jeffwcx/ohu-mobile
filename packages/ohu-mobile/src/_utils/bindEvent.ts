import { VueConstructor } from 'vue/types/umd';

export default function (
  instance: InstanceType<VueConstructor>,
  event: keyof WindowEventMap,
  handler: Function,
) {
  let registered = false;
  const listener = () => {
    handler();
  };
  const register = () => {
    window.addEventListener(event, listener);
    registered = true;
  };
  const unregister = () => {
    window.removeEventListener(event, listener);
    registered = false;
  };
  register();
  instance.$once('hook:beforeDestroy', unregister);
  instance.$once('hook:activated', () => {
    if (registered) return;
    register();
  });
  instance.$once('hook:deactivated', unregister);
}
