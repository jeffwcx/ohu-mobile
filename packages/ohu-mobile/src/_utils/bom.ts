export function getScrollTop(element: Element | Window) {
  if (element === window) {
    return Math.max(window.scrollY || 0, document.documentElement.scrollTop);
  }
  return (element as HTMLElement).scrollTop;
}

export function getClientHeight(element: Element | Window) {
  if (element === window) {
    return Math.max(document.documentElement.clientHeight, window.innerHeight);
  }
  return (element as HTMLElement).clientHeight;
}

export function getScrollHeight(element: HTMLElement) {
  return element.scrollHeight;
}

export function getTopOffsetOfWrapperAndInner(
  wrapper: Element | Window,
  inner: Element,
) {
  let top = 0;
  if (wrapper !== window) {
    const wrapperRect = (wrapper as Element).getBoundingClientRect();
    top = wrapperRect.top;
  }
  const innerRect = inner.getBoundingClientRect();
  return Math.abs(top - innerRect.top);
}

export function isElementSticky(scroller: Element | Window, el: Element) {
  let st = 0;
  if (scroller !== window) {
    st = (scroller as Element).getBoundingClientRect().top;
  }
  const { bottom } = el.getBoundingClientRect();
  return bottom - st > 0;
}

export function reachBottom(element: Element | Window, distance: number = 0) {
  const scrollElement =
    element === window ? document.body : (element as HTMLElement);
  const scrollHeight = getScrollHeight(scrollElement);
  return (
    scrollHeight - (getClientHeight(element) + getScrollTop(element)) <=
    distance
  );
}
