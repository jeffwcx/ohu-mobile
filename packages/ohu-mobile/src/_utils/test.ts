export async function wait(timeout: number = 0) {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(true);
    }, timeout);
  });
}
