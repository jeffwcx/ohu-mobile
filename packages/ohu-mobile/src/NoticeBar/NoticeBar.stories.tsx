import Vue from 'vue';

import NoticeBar from './index';
import './style';
import Card from '../Card';
import '../Card/style';
import Icon from '../Icon';
import '../Icon/style';
import Button from '../Button';
import '../Button/style';
import VueRouter from 'vue-router';
import { VolumeDownOutlined, LockFilled } from '@ohu-mobile/icons';

export default {
  title: 'Components/FeedBack/NoticeBar',
  parameters: {
    component: NoticeBar,
    options: {
      showPanel: true,
    },
  },
};

Vue.use(VueRouter);

const A = Vue.extend({
  render() {
    return (
      <div>
        <NoticeBar
          scrollable
          offset="50%"
          icon={VolumeDownOutlined}
          action="link"
        >
          初次使用，您有15分钟体验时间。For the first time, you have 15 minutes
          to experience.
        </NoticeBar>
        <Button type="primary" to="/b">
          打开B
        </Button>
      </div>
    );
  },
});

const B = Vue.extend({
  render() {
    return (
      <div class="demo">
        <Button type="primary" to="/a">
          打开A
        </Button>
      </div>
    );
  },
});

const router = new VueRouter({
  mode: 'hash',
  routes: [
    { path: '/a', component: A, alias: '/' },
    { path: '/b', component: B },
  ],
});

export const Basic = () =>
  Vue.extend({
    router,
    render() {
      return (
        <div>
          <Card divider shadow>
            <Card.Header>Basic</Card.Header>
            <NoticeBar>初次使用，您有15分钟体验时间</NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>icon</Card.Header>
            <NoticeBar icon={VolumeDownOutlined} speed={30}>
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>closable</Card.Header>
            <NoticeBar icon={VolumeDownOutlined} delay={2000} action="closable">
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>scrollable</Card.Header>
            <NoticeBar scrollable icon={VolumeDownOutlined} action="link">
              初次使用，您有15分钟体验时间。For the first time, you have 15
              minutes to experience.
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>offset 67%</Card.Header>
            <NoticeBar offset="67%" icon={VolumeDownOutlined} action="link">
              初次使用，您有15分钟体验时间。For the first time, you have 15
              minutes to experience.
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>offset 50% scrollable</Card.Header>
            <keep-alive>
              <router-view></router-view>
            </keep-alive>
          </Card>
          <Card divider shadow>
            <Card.Header>multiline</Card.Header>
            <NoticeBar icon={VolumeDownOutlined} action="closable" multiline>
              longtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtext
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>link</Card.Header>
            <NoticeBar icon={VolumeDownOutlined} action="link">
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>slot: icon</Card.Header>
            <NoticeBar icon={VolumeDownOutlined} action="link">
              <Icon type={LockFilled} slot="icon" color="red" />
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
        </div>
      );
    },
  });

export const Warning = () =>
  Vue.extend({
    render() {
      return (
        <div>
          <Card divider shadow>
            <Card.Header>Basic</Card.Header>
            <NoticeBar type="warning">初次使用，您有15分钟体验时间</NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>icon</Card.Header>
            <NoticeBar type="warning" icon={VolumeDownOutlined}>
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>closable</Card.Header>
            <NoticeBar
              type="warning"
              icon={VolumeDownOutlined}
              action="closable"
            >
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>multiline</Card.Header>
            <NoticeBar
              type="warning"
              icon={VolumeDownOutlined}
              action="closable"
              multiline
            >
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
          <Card divider shadow>
            <Card.Header>link</Card.Header>
            <NoticeBar type="warning" icon={VolumeDownOutlined} action="link">
              初次使用，您有15分钟体验时间, 初次使用，您有15分钟体验时间,
              初次使用，您有15分钟体验时间
            </NoticeBar>
          </Card>
        </div>
      );
    },
  });
