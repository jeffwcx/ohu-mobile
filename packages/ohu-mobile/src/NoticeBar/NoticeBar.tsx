import { defineComponent, props } from '../_utils/defineComponent';
import { NoticeBarProps, NoticeBarEvents, NoticeBarType, NoticeBarAction } from './types';
import { IconDef } from '../types';
import Icon from '../Icon';
import { CloseOutlined, ArrowRightSOutlined } from '@ohu-mobile/icons';
import { VNodeData } from 'vue/types/umd';

interface NoticeBarData {
  isShow: boolean;
  marqueeMoveDistance: number;
  marqueeWidth: number;
  marqueeContainerWidth: number;
  translateX: number | string;
  marqueeStyle: Partial<CSSStyleDeclaration>;
  getInitMarqueeStyle: () => Partial<CSSStyleDeclaration>;
}


export default defineComponent<NoticeBarProps, NoticeBarEvents, {}, NoticeBarData>('notice-bar').create({
  props: {
    text: String,
    icon: props<IconDef>(Object).optional,
    action: props.ofType<NoticeBarAction>().optional,
    type: props.ofType<NoticeBarType>().default('default'),
    multiline: props(Boolean).default(false),
    scrollable: props.ofType<Boolean | undefined>().optional,
    offset: props(String).default('100%'),
    delay: props(Number).default(1000),
    speed: props(Number).default(50),
  },
  data() {
    return {
      timeoutId: null as ReturnType<typeof setTimeout> | null,
      isShow: true,
      translateX: 0,
      marqueeMoveDistance: 0,
      marqueeWidth: 0,
      marqueeContainerWidth: 0,
      marqueeStyle: this.getInitMarqueeStyle(),
    };
  },
  computed: {
    isScroll() {
      if (this.scrollable !== undefined) return this.scrollable;
      return this.marqueeContainerWidth < this.marqueeWidth;
    },
    offsetLeft() {
      const match = this.offset.match(/(0|[^0]\d{0,1}|100)%?/);
      if (match && match[1]) {
        const value = +match[1];
        return value >= 0 && value <= 100 ? value : 0;
      }
      return 0;
    },
  },
  methods: {
    getInitMarqueeStyle() {
      return this.offset !== '100%' && !this.multiline
        ? { marginLeft: this.offset }
        : {};
    },
    handleAction(e: Event) {
      if (this.action === 'closable') {
        e.stopPropagation();
        this.isShow = false;
        this.$emit('close', e);
      }
    },
    setMarqueStyle(translateX?: number, duration?: number) {
      let translate = translateX !== undefined ? translateX : -this.marqueeWidth;
      const style: Partial<CSSStyleDeclaration> = {
        transitionTimingFunction: 'linear',
        transform: `translate3d(${translate}px, 0, 0)`,
        transitionDuration: duration === undefined
          ? `${this.marqueeMoveDistance / this.speed}s`
          : `${duration}s`,
      };
      this.marqueeStyle = style;
    },
    setMarqueeWidth() {
      const marquee = this.$refs.marquee as HTMLDivElement;
      const marqueeInner = this.$refs.marqueeInner as HTMLDivElement;
      if (!marqueeInner || !marquee) return;
      return {
        marqueeContainerWidth: marquee.clientWidth,
        marqueeWidth: marqueeInner.clientWidth,
      };
    },
    computeInitMoveDistance() {
      return this.marqueeContainerWidth * (this.offsetLeft / 100) + this.marqueeWidth;
    },
    initMarquee() {
      if (this.timeoutId) clearTimeout(this.timeoutId);
      Object.assign(this, this.setMarqueeWidth());
      if (!this.isScroll) return;
      this.marqueeMoveDistance = this.computeInitMoveDistance();
      this.timeoutId = setTimeout(() => {
        this.setMarqueStyle();
        this.timeoutId = null;
      }, this.delay);
      this.$on('hook:beforeDestroy', () => {
        this.timeoutId && clearTimeout(this.timeoutId);
      });
      this.$on('hook:activated', () => {
        this.resetMarquee();
      });
      this.$on('hook:deactivated', () => {
        this.timeoutId && clearTimeout(this.timeoutId);
      });
    },
    resetMarquee() {
      Object.assign(this, this.setMarqueeWidth());
      if (!this.isScroll) {
        return this.marqueeStyle = this.getInitMarqueeStyle();
      }
      this.marqueeMoveDistance = this.marqueeContainerWidth + this.marqueeWidth;
      this.setMarqueStyle(this.marqueeContainerWidth, 0);
      setTimeout(() => {
        this.setMarqueStyle();
      }, 300);
    },
  },
  mounted() {
    this.initMarquee();
  },
  render() {
    const {
      $slots, isShow,
      icon, action, multiline, type,
      isScroll, text,
    } = this;
    const root = this.root()
      .has([
        icon && 'icon',
        action && 'action',
      ])
      .is(type);
    const textNode: VNodeData = {
      ref: 'marquee',
      class: root.element('text')
        .is([
          multiline ? 'wrap' : 'inline',
          isScroll && 'scrollable',
        ]),
    };
    if (isScroll) {
      textNode.on = {
        transitionend: this.resetMarquee,
      };
    }
    return (
      <div v-show={isShow}
        class={root}
        onClick={(e) => this.$emit('click', e)}>
        {
          $slots.icon
            ? (
              <div class={root.element('icon')}>
                {$slots.icon}
              </div>
            )
            : (
              icon
              &&
              <div class={root.element('icon')}>
                <Icon type={icon} />
              </div>
            )
        }
        <div {...textNode}>
          <div ref="marqueeInner" style={this.marqueeStyle}>{text || $slots.default}</div>
        </div>
        {
          action
          &&
          <div role="button" class={root.element('action')} onClick={this.handleAction}>
            <Icon type={action === 'closable' ? CloseOutlined : ArrowRightSOutlined} />
          </div>
        }
      </div>
    );
  },
});
