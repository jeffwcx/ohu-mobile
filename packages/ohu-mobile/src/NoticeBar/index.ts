import NoticeBar from './NoticeBar';

export default NoticeBar;

export * from './types';
