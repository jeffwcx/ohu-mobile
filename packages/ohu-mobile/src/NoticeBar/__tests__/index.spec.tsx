import { VolumeDownOutlined } from '@ohu-mobile/icons';
import { config, mount } from '@vue/test-utils';
import NoticeBar from '..';
import { wait } from '../../_utils/test';

describe('NoticeBar', () => {
  // events
  it('close event', async () => {
    const onClose = jest.fn();
    const onClick = jest.fn();
    const wrapper = mount(NoticeBar, {
      propsData: {
        action: 'closable',
        text: 'Text',
      },
      listeners: {
        close: onClose,
        click: onClick,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
    await wrapper.find('[role=button]').trigger('click');
    expect(onClose).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledTimes(0);
    expect(wrapper.isVisible()).toBeFalsy();
    await wrapper.trigger('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  // props
  it('multiline', async () => {
    const wrapper = mount(NoticeBar, {
      propsData: {
        text: 'longtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtext',
      },
    });
    expect(wrapper.find('.ohu-notice-bar__text').classes('is-inline')).toBeTruthy();
    await wrapper.setProps({ multiline: true });
    expect(wrapper.find('.ohu-notice-bar__text').classes('is-wrap')).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('set icon', () => {
    const wrapper = mount(NoticeBar, {
      propsData: {
        icon: VolumeDownOutlined,
        text: 'Text'
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('action', () => {
    const wrapper = mount(NoticeBar, {
      propsData: {
        action: 'link'
      },
      slots: {
        icon: 'S',
        default: 'Text'
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  config.showDeprecationWarnings = false;
  it('scrollable', async () => {
    const setW = jest.fn().mockImplementation(() => {
      wrapper.setData({
        marqueeWidth: 543,
        marqueeContainerWidth: 524,
      });
    });
    const wrapper = mount(NoticeBar, {
      propsData: {
        text: 'longtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtext',
        scrollable: true,
      },
      methods: {
        setMarqueeWidth() {
          return {
            marqueeWidth: 543,
            marqueeContainerWidth: 524,
          };
        },
      },
    });
    await wait(1000);
    const inner = wrapper.find('.ohu-notice-bar__text div');
    const transformText = inner.element.style['transform'];
    expect(transformText).toBe('translateX(-543px)');
    await inner.trigger('transitionend');
    expect(inner.element.style['transform']).toBe('translateX(524px)');
    await wait(300);
    expect(inner.element.style['transform']).toBe('translateX(-543px)');
  });

});
