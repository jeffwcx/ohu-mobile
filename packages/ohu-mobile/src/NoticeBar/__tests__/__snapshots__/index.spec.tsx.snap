// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeBar action 1`] = `
<div class="ohu-notice-bar has-action is-default">
  <div class="ohu-notice-bar__icon">S</div>
  <div class="ohu-notice-bar__text is-inline">
    <div>Text</div>
  </div>
  <div role="button" class="ohu-notice-bar__action"><svg viewBox="0 0 24 24" aria-hidden="true" tabindex="-1" title="arrow-right-s" role="presentation" focusable="false" class="ohu-icon">
      <path fill="none" d="M0 0h24v24H0z"></path>
      <path d="M13.172 12l-4.95-4.95 1.414-1.414L16 12l-6.364 6.364-1.414-1.414z"></path>
    </svg></div>
</div>
`;

exports[`NoticeBar close event 1`] = `
<div class="ohu-notice-bar has-action is-default">
  <div class="ohu-notice-bar__text is-inline">
    <div>Text</div>
  </div>
  <div role="button" class="ohu-notice-bar__action"><svg viewBox="0 0 24 24" aria-hidden="true" tabindex="-1" title="close" role="presentation" focusable="false" class="ohu-icon">
      <path fill="none" d="M0 0h24v24H0z"></path>
      <path d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636z"></path>
    </svg></div>
</div>
`;

exports[`NoticeBar multiline 1`] = `
<div class="ohu-notice-bar is-default">
  <div class="ohu-notice-bar__text is-wrap">
    <div>longtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtextlongtext</div>
  </div>
</div>
`;

exports[`NoticeBar set icon 1`] = `
<div class="ohu-notice-bar has-icon is-default">
  <div class="ohu-notice-bar__icon"><svg viewBox="0 0 24 24" aria-hidden="true" tabindex="-1" title="volume-down" role="presentation" focusable="false" class="ohu-icon">
      <path fill="none" d="M0 0h24v24H0z"></path>
      <path d="M13 7.22L9.603 10H6v4h3.603L13 16.78V7.22zM8.889 16H5a1 1 0 0 1-1-1V9a1 1 0 0 1 1-1h3.889l5.294-4.332a.5.5 0 0 1 .817.387v15.89a.5.5 0 0 1-.817.387L8.89 16zm9.974.591l-1.422-1.422A3.993 3.993 0 0 0 19 12c0-1.43-.75-2.685-1.88-3.392l1.439-1.439A5.991 5.991 0 0 1 21 12c0 1.842-.83 3.49-2.137 4.591z"></path>
    </svg></div>
  <div class="ohu-notice-bar__text is-inline">
    <div>Text</div>
  </div>
</div>
`;
